examples/fluids/include/mat-ceed-impl.h

```diff
@@ -0,0 +1,69 @@
+#ifndef MAT_CEED_IMPL_H
+#define MAT_CEED_IMPL_H
+
+#include <ceed.h>
+#include <petscdm.h>
+#include <petscmat.h>
+
+#if defined(__clang_analyzer__)
+#define MATCEED_EXTERN extern
+#elif defined(__cplusplus)
+#define MATCEED_EXTERN extern "C"
+#else
+#define MATCEED_EXTERN extern
+#endif
+
+#if defined(__clang_analyzer__)
+#define MATCEED_INTERN
+#else
+#define MATCEED_INTERN MATCEED_EXTERN __attribute__((visibility("hidden")))
+#endif
+
+/**
+  @brief Calls a libCEED function and then checks the resulting error code.
+  If the error code is non-zero, then a PETSc error is set with the libCEED error message.
+**/
+#define PetscCeedCall(ceed_, ...)                                   \
```

Comment:
Note - renamed `PetscCallCeed` in Ratel to match PETSc conventions

Reply:
Yep, I saw that. Figured I'd do that and the other changes from https://gitlab.com/micromorph/ratel/-/merge_requests/794 in a later PR. But changing this one isn't a big deal.

Reply:
Cool. I figure there will be some minor effort keeping these two in sync, but nothing too bad

Reply:
Yep, that was my plan.